src/gallia/services/uds/ecu.py

```diff
@@ -209,27 +209,29 @@ async def callback() -> None:
         self.state.reset()
         return True
 
-    async def leave_session(self, level: int, config: UDSRequestConfig | None = None) -> bool:
+    async def leave_session(
+        self,
+        level: int,
+        config: UDSRequestConfig | None = None,
+        sleep: int | None = None,
+    ) -> bool:
         """leave_session() is a hook which can be called explicitly by a
         scanner when a session is to be disabled. Use this hook if resetting
         the ECU is required, e.g. when disabling the programming session.
-
-        Args:
-            uds: The UDSClient class where this hook is embedded. The caller typically
-                 calls this function with `self` as the first argument.
-            session: The desired session identifier.
-        Returns:
-            True on success, False on error.
         """
+        power_cycle_args = {}
+        if sleep is not None:
+            power_cycle_args["sleep"] = sleep
+
         resp: service.UDSResponse = await self.ecu_reset(0x01)
         if isinstance(resp, service.NegativeResponse):
-            await self.power_cycle()
+            await self.power_cycle(**power_cycle_args)
             await self.reconnect()
         await self.wait_for_ecu()
 
         resp = await self.set_session(0x01, config=config)
         if isinstance(resp, service.NegativeResponse):
-            await self.power_cycle()
+            await self.power_cycle(**power_cycle_args)
```

Comment:
Why not just `self.power_cycle(sleep)`?

Reply:
To allow for using the default sleep time defined in `power_cycle()` without repeating its definition in `leave_session()`. The latter might be easily overseen, should the definition in `power_cycle()` be updated in the future.

Reply:
When you use `**power_cycle_args` it does not get catched by `mypy`. When you use plain `sleep` as an argument it is all statically typed.